### ISSUE TITLE
Bring python_package_template up to date

### ISSUE THREAD

**Issue by heather999** [LSSTDESC/ReleaseManager#1]
This could be very useful, need to get this functioning again, and bring up to date with GitHub Actions
https://github.com/LSSTDESC/python_package_template

**Comment by eacharles** [comment]
Got this at least building in a fork: https://github.com/eacharles/python_package_template

We should think about the things we want / don't want.

My thoughts

Want:
- Optional simple github actions for CI 
  - flake8 / pylint
  - unit test / coverage
- Optional github action for pypi deploy
- Optional to use for personal  repos (i.e., not DESC copyrighted)
- setuptools_scm and setuptools_scm_git_version for version control
- readthedocs docs w/ sphinx
- setup.cfg and setup.py 

Don't want:
- Travis
- bumpversion stuff (use github release creation instead)

**Comment by heather999** [comment]
Cool - thanks for taking a look! That all sounds reasonable.  Providing templates for things we want to encourage (linting, unit testing/coverage..) will help developers use those tools.  

If something is not listed as "optional" above, what does that imply? (I'm wondering about readthedocs/sphinx) - I'm all for it, but if a project isn't ready for something yet - they should be able to turn it off while they get started? 

Do we want to think about moving to [pyproject.toml ](https://github.com/pypa/setuptools_scm)now as a replacement to setup.py & setup.cfg?  Not an expert on this.. but if we're starting fresh, maybe it's a good time to make such a change?

**Comment by eacharles** [comment]
It just means I forgot to type "optional", and yeah, we should probably move to pyproject.toml, though we should maintain setup.py as it is needed for "editable" build.  (I think that means what you get with 'python setup.py develop", correct?)

**Comment by heather999** [comment]
Yes, I think we just need a shim setup.py in this case to allow the use of `python setup.py develop`, something like:
```
#!/usr/bin/env python

import setuptools

if __name__ == "__main__":
    setuptools.setup()
```

My read of https://www.python.org/dev/peps/pep-0660/ suggests there may be further changes coming to allow pyproject.toml to support editable builds - but it just seems prudent to support setup.py for this use case for now.

**Comment by heather999** [comment]
Bringing some helpful comments from Eric over from another context:
"qp / tables_io have simple linting, unit tests, and a very strong coverage requirement.   Probably want some discussion of what we want in the template.  E.g., probably not a github action that requires 100% coverage to start with."
